### ISSUE TITLE
Are the `#![feature(layout_for_ptr)]` functions sound to call on invalid data pointers?

### ISSUE THREAD

**Issue by repnop** [rust-lang/rust#86602]
Reading the documentation for the [`#![feature(layout_for_ptr)]`](https://github.com/rust-lang/rust/issues/69835) functions, it doesn't mention if it accepts invalid data pointer values (e.g. a pointer created with `core::ptr::null()`), but seems to suggest that it does:
> If `T` is `Sized`, this function is always safe to call.

but there's a comment mentioning that the pointer must be valid inside the functions, e.g.:
https://github.com/rust-lang/rust/blob/456a03227e3c81a51631f87ec80cac301e5fa6d7/library/core/src/mem/mod.rs#L375-L378

I would think that the answer is yes since its operating on the metadata of the pointer and not the pointer value itself (and with my limited testing, this seems like the current behavior). However, in either case I think its worth explicitly mentioning the validity of the pointer in the documentation for the functions.

**Comment by asquared31415** [comment]
[playground link](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=9085236f90e3f42a9cd3b02b4478e5f6) demonstrating that the metadata is read correctly despite the pointer being invalid in the traditional sense

**Comment by RalfJung** [comment]
> but there's a comment mentioning that the pointer must be valid inside the functions, e.g.:

Yes, the caller must provide a valid *raw pointer*. Which values are valid for which type is [documented in the reference](https://doc.rust-lang.org/reference/behavior-considered-undefined.html).
